packages/steiger/src/app.ts

```diff
@@ -21,25 +21,22 @@ async function runRules({ vfs, rules }: { vfs: Folder; rules: Array<Rule> }) {
   const vfsWithoutGlobalIgnores = removeGlobalIgnoreFromVfs(vfs, getGlobalIgnores())
 
   const ruleResults = await Promise.all(rules.map((rule) => runRule(vfsWithoutGlobalIgnores, rule)))
-  return ruleResults.flatMap((r, ruleResultsIndex) => {
-    const { diagnostics } = r
-    if (diagnostics.length === 0) {
-      return []
-    }
```

Comment:
note: I removed this part while I was adding the sorting because I found several return statements to be confusing and because I believe this isn't that impactful from the performance perspective. The empty array will simply get processed as any other array, with no work done on it